cli/update.go

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) 2016,2017 Docker, Inc.
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/docker/go-units"
+	vc "github.com/kata-containers/runtime/virtcontainers"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/urfave/cli"
+)
+
+func i64Ptr(i int64) *int64   { return &i }
+func u64Ptr(i uint64) *uint64 { return &i }
+func u16Ptr(i uint16) *uint16 { return &i }
+
+var updateCLICommand = cli.Command{
+	Name:      "update",
+	Usage:     "update container resource constraints",
+	ArgsUsage: `<container-id>`,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "resources, r",
+			Value: "",
+			Usage: `path to the file containing the resources to update or '-' to read from the standard input
+
+The accepted format is as follow (unchanged values can be omitted):
+
+{
+  "memory": {
+    "limit": 0,
+    "reservation": 0,
+    "swap": 0,
+    "kernel": 0,
+    "kernelTCP": 0
+  },
+  "cpu": {
+    "shares": 0,
+    "quota": 0,
+    "period": 0,
+    "realtimeRuntime": 0,
+    "realtimePeriod": 0,
+    "cpus": "",
+    "mems": ""
+  },
+  "blockIO": {
+    "weight": 0
+  }
```

Comment:
missing pids?

Reply:
I don't think so, it's base on runC implementation

Reply:
It's a bug in runc? `--pids-limit` and `--l3-cache-schema` were added after the initial `runc update` implementation and it seems the developer forgot to add them to the resources file.

Reply:
ref opencontainers/runc/issues/1793

Reply:
We do read pid limit at https://github.com/kata-containers/runtime/pull/270/files#diff-1cc31db47bce8b7af5fda9d984c048e6R162. So I think we should at least add pids here. l3-cache-schema may be left out since it is not part of `specs.LinuxResources` anyway.

Reply:
good catch!